drivers/gpio/gpio_nrfx.c

```diff
@@ -513,7 +515,9 @@ static int gpio_nrfx_init(struct device *port)
 #define GPIO_NRF_DEVICE(id)						\
 	static const struct gpio_nrfx_cfg gpio_nrfx_p##id##_cfg = {	\
 		.common = {						\
-			.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_NGPIOS(DT_INST_##id##_NORDIC_NRF_GPIO_NGPIOS), \
+			.port_pin_mask =				\
+				GPIO_PORT_PIN_MASK_FROM_NGPIOS(		\
+					DT_INST_PROP(id, ngpios)),	\
```

Comment:
This is wrong. It was also wrong before this change. `id` here is the hardware instance index, not the DT instance index,
so it should be rather `DT_PROP(DT_NODELABEL(gpio##id), ngpios)`.

Reply:
Nice catch.

Would you prefer I just convert all the nrf drivers to the new devicetree API so we catch changes like this in one go, and don't have to worry about it more?

Reply:
Sounds reasonable.

Reply:
>> Would you prefer I just convert all the nrf drivers to the new devicetree API [...]
>
> Sounds reasonable.

OK, I'm following up in separate per-driver pull requests. This will hopefully make the reviews smaller to do, and allow us to merge drivers that are OK faster, since an issue in one driver won't block the conversion of another.